### PR TITLE
Feat/linking users with timesheets

### PR DESCRIPTION
**Pull request by oskarikerppo**

### CODE REVIEW

**Comment by oskarikerppo** [comment]
Screenshot from 2021-12-01 16-48-21

File: components/timesheets/TimesheetTable.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                <Button colorScheme="blue" onClick={onOpen}>
+                    Add User
+                </Button>
+            </Flex>
+            <Modal isOpen={isOpen} onClose={onClose}>
+                <ModalOverlay />
+                <ModalContent>
+                    <ModalHeader>Add user to project</ModalHeader>
+                    <ModalCloseButton />
+                    <FormControl>
```
Comment (robertsiipola) [inline components/timesheets/TimesheetTable.tsx:167]: Please add left and right padding to this form. 0.5 rem should be fine on both sides.

**Review by robertsiipola — CHANGES_REQUESTED** [review]

**Comment by LeadingMoominExpert** [comment]
> The button "x" in the table doesn't do anything yet. TimesheetDTO needs a field "status" that can be set to "archived" first. Also there needs to be some check either in the backend or frontend that a user is not added multiple times, a users total allocation is not over 100, etc.

Backend validation for correct allocation already exists: https://github.com/three-consulting/epoc-backend/blob/main/src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt#L13

For the combination of employee and project to be unique on timesheet, we can do a multi parameter constraint like https://stackoverflow.com/a/2772510/7990795

File: components/timesheets/TimesheetTable.tsx (hunk 1 above)
Comment (oskarikerppo) [reply components/timesheets/TimesheetTable.tsx:167]: Screenshot from 2021-12-08 10-44-52
added in 727f11d

**Review by robertsiipola — APPROVED** [review]

**Pull request merged**